tools/rust_api/src/value.rs

```diff
@@ -0,0 +1,879 @@
+        convert_node: Value::Node(NodeVal::new(InternalID { table_id: 0, offset: 0 }, "Test Label".to_string())),
+        convert_rel: Value::Rel(RelVal::new(InternalID { table_id: 0, offset: 0 }, InternalID { table_id: 1, offset: 0 }, "Test Label".to_string())),
+    }
+
+    display_tests! {
+        display_var_list: Value::VarList(LogicalType::String, vec!["Alice".into(), "Bob".into()]),
+        display_var_list_empty: Value::VarList(LogicalType::String, vec![]),
+        display_fixed_list: Value::FixedList(LogicalType::String, vec!["Alice".into(), "Bob".into()]),
+        display_int16: Value::Int16(1),
+        display_int32: Value::Int32(2),
+        display_int64: Value::Int64(3),
+        // Float, doble, interval and timestamp have display differences which we probably don't want to
+        // reconcile
+        display_date: Value::Date(date!(2023-06-13)),
+        display_string: Value::String("Hello World".to_string()),
+        display_bool: Value::Bool(false),
+        display_null: Value::Null(LogicalType::VarList {
+            child_type: Box::new(LogicalType::FixedList { child_type: Box::new(LogicalType::Int16), num_elements: 3 })
+        }),
+        display_struct: Value::Struct(vec![("NAME".to_string(), "Alice".into()), ("AGE".to_string(), 25.into())]),
+        display_internal_id: Value::InternalID(InternalID { table_id: 0, offset: 0 }),
+        display_node: Value::Node(NodeVal::new(InternalID { table_id: 0, offset: 0 }, "Test Label".to_string())),
+        display_rel: Value::Rel(RelVal::new(InternalID { table_id: 0, offset: 0 }, InternalID { table_id: 1, offset: 0 }, "Test Label".to_string())),
+    }
+
+    database_tests! {
+        // Passing these values as arguments is not yet implemented in kuzu:
+        // db_struct:
+        //    Value::Struct(vec![("item".to_string(), "Knife".into()), ("count".to_string(), 1.into())]),
+        //    "STRUCT(item STRING, count INT32)",
+        // db_fixed_list: Value::FixedList(LogicalType::String, vec!["Alice".into(), "Bob".into()]), "STRING[2]",
+        db_var_list_string: Value::VarList(LogicalType::String, vec!["Alice".into(), "Bob".into()]), "STRING[]",
+        db_var_list_int: Value::VarList(LogicalType::Int64, vec![0i64.into(), 1i64.into(), 2i64.into()]), "INT64[]",
+        db_int16: Value::Int16(1), "INT16",
+        db_int32: Value::Int32(2), "INT32",
+        db_int64: Value::Int64(3), "INT64",
+        db_float: Value::Float(4.), "FLOAT",
+        db_double: Value::Double(5.), "DOUBLE",
+        db_timestamp: Value::Timestamp(datetime!(2023-06-13 11:25:30 UTC)), "TIMESTAMP",
+        db_date: Value::Date(date!(2023-06-13)), "DATE",
+        db_interval: Value::Interval(time::Duration::weeks(200)), "INTERVAL",
+        db_string: Value::String("Hello World".to_string()), "STRING",
+        db_bool: Value::Bool(true), "BOOLEAN",
+        // Causes buffer manager failure
+        // db_null_string: Value::Null(LogicalType::String), "STRING",
```

Comment:
As we discussed, we currently don't support giving null/list values to parameters of preparedStatement.
you can remove this.

Reply:
Done